File: agent-cpp/src/agent/player_info.cc
#include "player_info.h"

#include <fmt/ranges.h>

#include <magic_enum.hpp>
#include <string>

namespace thuai7_agent {

auto format_as(ArmorKind object) -> std::string {
  return std::string(magic_enum::enum_name(object));
}

auto format_as(FirearmKind object) -> std::string {
  return std::string(magic_enum::enum_name(object));
}

auto format_as(MedicineKind object) -> std::string {
  return std::string(magic_enum::enum_name(object));
}

auto format_as(ItemKind object) -> std::string {
  return std::string(magic_enum::enum_name(object));
}

auto format_as(Item const& object) -> std::string {
  return fmt::format("Item{{kind: {}, count: {}}}", object.kind, object.count);
}

auto format_as(PlayerInfo const& object) -> std::string {
  return fmt::format(
      "PlayerInfo{{id: {}, armor: {}, current_armor_health: {}, health: {}, "
      "speed: {}, firearm: {}, firearms_pool:{}, range: {}, "
      "position: {}, inventory: {}}}",
      object.id, object.armor, object.current_armor_health, object.health,
      object.speed, object.firearm, object.firearms_pool, object.range,
      object.position, object.inventory);
}

}  // namespace thuai7_agent

File: agent-python/agent/player_info.py
from typing import List, Literal

from .position import Position

ArmorKind = Literal[
    "NO_ARMOR",
    "PRIMARY_ARMOR",
    "PREMIUM_ARMOR",
]

FirearmKind = Literal[
    "FIST",
    "S686",
    "M16",
    "AWM",
    "VECTOR",
]

ItemKind = Literal[
    "BANDAGE",
    "FIRST_AID",
    "BULLET",
    "GRENADE",
]


class Item:
    def __init__(self, kind: ItemKind, count: int):
        self.kind = kind
        self.count = count


class PlayerInfo:
    def __init__(
        self,
        id: int,
        armor: ArmorKind,
        current_armor_health: float,
        health: int,
        speed: float,
        firearm: FirearmKind,
        firearms_pool: List[FirearmKind],
        range: float,
        position: Position[float],
        inventory: List[Item],
    ):
        self.id = id
        self.armor = armor
        self.current_armor_health = current_armor_health
        self.health = health
        self.speed = speed
        self.firearm = firearm
        self.firearms_pool = firearms_pool
        self.range = range
        self.position = position
        self.inventory = inventory

    def __str__(self) -> str:
        return f"PlayerInfo{{id: {self.id}, armor: {self.armor}, current_armor_health:{self.current_armor_health}, health: {self.health}, speed: {self.speed}, firearm: {self.firearm}, firearms_pool:{self.firearms_pool}, range: {self.range}, position: {self.position}, inventory: {self.inventory}}}"
